tests/sqlparser_mysql.rs

```diff
@@ -222,6 +222,39 @@ fn parse_escaped_string() {
     }
 }
 
+#[test]
+fn parse_escaped_double_quoted_string() {
+    let sql = r#"SELECT "I\"m fine""#;
+
+    let stmt = mysql().one_statement_parses_to(sql, "");
+
+    match stmt {
+        Statement::Query(query) => match query.body {
+            SetExpr::Select(value) => {
+                let expr = expr_from_projection(only(&value.projection));
+                assert_eq!(
+                    *expr,
+                    Expr::Value(Value::DoubleQuotedString("I\"m fine".to_string()))
+                );
+            }
+            _ => unreachable!(),
+        },
+        _ => unreachable!(),
+    };
+
+    let sql = r#"SELECT "I""m fine""#;
+
+    let projection = mysql().verified_only_select(sql).projection;
+    let item = projection.get(0).unwrap();
+
+    match &item {
+        SelectItem::UnnamedExpr(Expr::Value(value)) => {
+            assert_eq!(*value, Value::DoubleQuotedString("I\"m fine".to_string()));
+        }
+        _ => unreachable!(),
+    }
+}
+
```

Comment:
I think we should also add tests for the following cases
1. single quoted strings elsewhere in the query (not just the select list)
2. a double quoted string with no escapes
3. A double quoted string with at least two escaped `"`
4. An incorrectly quoted string (e.g. `SELECT "f"" from foo`



```
mysql> select * from bar where x != "foo";
+------+
| x    |
+------+
|    1 |
+------+
1 row in set, 1 warning (0.00 sec)

```

Reply:
Also a test that `SELECT "foo" ` is treated as `SELECT foo` in other dialects (like postgres)